src/doc/rustdoc/src/the-doc-attribute.md

```diff
@@ -214,3 +214,44 @@ the `strip-hidden` pass is removed.
 Since primitive types are defined in the compiler, there's no place to attach documentation
 attributes. This attribute is used by the standard library to provide a way to generate
 documentation for primitive types.
+
+## `#[doc(cfg(...))]` / `#[cfg(rustdoc)]`: Documenting platform-/feature-specific information
+
+Because of the way Rustdoc documents a crate, the documentation it creates is specific to the target
+rustc compiles for. Anything that's specific to any other target is dropped via `#[cfg]` attribute
+processing early in the compilation process. However, Rustdoc has a trick up its sleeve to handle
+platform-specific code if it *does* receive it.
+
+Because Rustdoc doesn't need to fully compile a crate to binary, it replaces function bodies with
+`loop {}` to prevent having to process more than necessary. This means that any code within a
```

Comment:
How does this cope with functions that have `-> impl Trait` where `!` does not implement `Trait`?

Reply:
cc @QuietMisdreavus

Reply:
There is currently an exception for `-> impl Trait`: https://github.com/rust-lang/rust/blob/26451ef7b5e00887dc8f27717ff34262df23d655/src/librustc_interface/util.rs#L668-L673

This means that `#[doc(cfg(...))]` doesn't work for those functions yet so I don't think this feature is ready for stabilization yet.

Reply:
```rust
//  * `impl Trait`, due to issue #43869 that functions returning impl Trait cannot be diverging.
//    Solving this may require `!` to implement every trait, which relies on the an even more
//    ambitious form of the closed RFC #1637. See also [#34511].
```
This bit seems unlikely from my POV as a T-Lang member... cc @rust-lang/lang if y'all have thoughts...

Reply:
I'm with @ollie27 here, i had assumed the reason we couldn't stabilize `doc(cfg)` yet was because the implementation didn't work for all situations.